### ISSUE TITLE
Unbound TypeVar apparently undetected in function overload

### ISSUE THREAD

**Issue by AlexWaygood** [microsoft/pyright#3288]
I just merged a [PR over at typeshed](https://github.com/python/typeshed/pull/7578) to improve the stub for `builtins.sum`. The [first draft](https://github.com/python/typeshed/pull/7578/commits/f41f186c4ec37b6ad712013eca1d6f7962e1d4cf) of the PR proposed  defining `sum` like this:

```python
import sys
from typing import *

class _SupportsSum(Protocol):
     def __add__(self, __x: Any) -> Any: ...

 _SumT = TypeVar("_SumT", bound=_SupportsSum)
 _SumS = TypeVar("_SumS", bound=_SupportsSum)

 @overload
 def sum(__iterable: Iterable[_SumT]) -> Any | Literal[0]: ...

if sys.version_info >= (3, 8):
     @overload
     def sum(__iterable: Iterable[_SumT], start: _SumS) -> _SumT | _SumS: ...

 else:
     @overload
     def sum(__iterable: Iterable[_SumT], __start: _SumS) -> _SumT | _SumS: ...
```

I think this should have triggered pyright's `reportInvalidTypeVarUse` check, as the TypeVar in the first overload is only used once. However, pyright reported no errors in our CI check.

(The PR was revised to correct this mistake before it was merged.)

**Comment by erictraut** [comment]
The `reportInvalidTypeVarUse` exempts constrained and bound TypeVars because those have legitimate (or at least well-defined) uses when only one instance appears within a signature. Since this is a bound TypeVar, the check didn't emit a diagnostic in this case.

**Comment by AlexWaygood** [comment]
Thanks!
